File: src/test/com/intellij/lang/jsgraphql/validation/GraphQLSuppressTest.java
package com.intellij.lang.jsgraphql.validation;

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase;
import org.jetbrains.annotations.NotNull;

public class GraphQLSuppressTest extends GraphQLTestCaseBase {
    @Override
    protected @NotNull String getBasePath() {
        return "/validation/suppress";
    }

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        enableAllInspections();
    }

    public void testSuppressOnDefinition() {
        doHighlightingTest();
    }

    public void testSuppressOnFile() {
        doHighlightingTest();
    }

    public void testSuppressOnStatementInjected() {
        doHighlightingTest("js");
    }

    public void testSuppressOnFileInjected() {
        doHighlightingTest("js");
    }
}

File: tests/testSrc/com/intellij/lang/jsgraphql/validation/GraphQLSuppressTest.kt
package com.intellij.lang.jsgraphql.validation

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase
import com.intellij.openapi.progress.runBlockingCancellable

class GraphQLSuppressTest : GraphQLTestCaseBase() {
  override fun getBasePath(): String {
    return "/validation/suppress"
  }

  override fun setUp() {
    super.setUp()
    enableAllInspections()
  }

  fun testSuppressOnDefinition() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testSuppressOnFile() = runBlockingCancellable {
    doHighlightingTest()
  }

  fun testSuppressOnStatementInjected() = runBlockingCancellable {
    doHighlightingTest("js")
  }

  fun testSuppressOnFileInjected() = runBlockingCancellable {
    doHighlightingTest("js")
  }
}
